### ISSUE TITLE
[Counterspell] not interrupt Evocation (BB #475)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#475]
This issue was migrated from bitbucket.
**Original Reporter:** Alex_Step
**Original Date:** 08.01.2014 08:40:42 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475

<hr>

Using [Counterspell] not interrupt Evocation mage spell.. 

Description [Counterspell]: Counters the enemy's spellcast, preventing spell from that school of magic from being cast for 8 sec.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.01.2014 18:08:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475/#comment-7642198

<hr>

Until we do a rewrite of the function that handles spell interruption the below fix will work as a temporarily solution.

Add to LoadSpellCustomAttr

```
#!c++

        case 12051: // Evocation - now we can interrupt this
            spellInfo->InterruptFlags |= SPELL_INTERRUPT_FLAG_INTERRUPT;
            break;
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Alex_Step
**Original Date:** 08.01.2014 18:37:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475/#comment-7642551

<hr>

thank you, now this is works for me

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 08.01.2014 18:54:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475/#comment-7642809

<hr>

:) I'll be talking with Yooki and looking at a better way to handle this tonight. It could affect some other spells as well.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 10.01.2014 03:03:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475/#comment-7664852

<hr>

Proper solution has been added to core. Resolved.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 10.01.2014 09:07:30 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475/#comment-7667537

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 11.01.2014 08:57:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/475/#comment-7680632

<hr>

Resolved issue #475 - Counterspell doesn't work:

Implemented a cleaner way to determine if a spell should be interrupted based on channel interrupt flags.
Evocation will now be interrupted by Counterspell as intended. All other spells that were affected by this
issue will now hopefully be rid of this problem.

This patch is property of LordUsagi and all credit goes to him.

→ <<cset f7ebdffe97d8>>
